cmd/kubelet/app/server.go

```diff
@@ -418,11 +425,10 @@ func InitializeTLS(s *options.KubeletServer) (*server.TLSOptions, error) {
 	if s.TLSCertFile == "" && s.TLSPrivateKeyFile == "" {
 		s.TLSCertFile = path.Join(s.CertDirectory, "kubelet.crt")
 		s.TLSPrivateKeyFile = path.Join(s.CertDirectory, "kubelet.key")
-		if crypto.ShouldGenSelfSignedCerts(s.TLSCertFile, s.TLSPrivateKeyFile) {
-			if err := crypto.GenerateSelfSignedCert(nodeutil.GetHostname(s.HostnameOverride), s.TLSCertFile, s.TLSPrivateKeyFile, nil, nil); err != nil {
-				return nil, fmt.Errorf("unable to generate self signed cert: %v", err)
+		if !crypto.FoundCertOrKey(s.TLSCertFile, s.TLSPrivateKeyFile) {
```

Comment:
@liggitt I should comment that the cert/key files are preferred to stay in some tmpfs(the default behavior now) so when the node reboots, they blow away automatically.

We could have a follow up PR to check for the existing certs/keys, or if you want, I can make an additional commit to check the cert/key.